flink-core/src/main/java/org/apache/flink/core/plugin/Plugin.java

```diff
@@ -21,13 +21,26 @@
 import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.configuration.Configuration;
 
+import javax.annotation.Nullable;
+
 /**
  * Interface for plugins. Plugins typically extend this interface in their SPI and the concrete implementations of a
  * service then implement the SPI contract.
  */
 @PublicEvolving
 public interface Plugin {
 
+	/**
+	 * Helper method to get the class loader used to load the plugin. This may be needed for some plugins that use
+	 * dynamic class loading afterwards the plugin was loaded.
+	 *
+	 * @return the class loader used to load the plugin. {@code null} value may represent the bootstrap class loader.
```

Comment:
(re `null`) - I think this is impossible to happen. How could bootstrap class loader be responsible for loading a `Plugin` class? It would have to be provided by rt.jar

If I'm right, I would remove `@Nullable` annotation and add a checkState here `this.getClass().getClassLoader() != null`, to avoid `null` handling branches.